src/nodes.coffee

```diff
@@ -4102,6 +4136,19 @@ exports.Code = class Code extends Base
       @name.error 'Class constructor may not be async'       if @isAsync
       @name.error 'Class constructor may not be a generator' if @isGenerator
 
+  disallowLoneExpansionAndMultipleSplats: ->
+    haveSplatParam   = no
+    for param, i in @params
+      # Was `...` used with this parameter? (Only one such parameter is allowed
+      # per function.) Splat/expansion parameters cannot have default values,
+      # so we need not worry about that.
+      if param.splat or param instanceof Expansion
+        if haveSplatParam
+          param.error 'only one splat or expansion parameter is allowed per function definition'
+        else if param instanceof Expansion and @params.length is 1
+          param.error 'an expansion parameter cannot be the only parameter in a function definition'
+        haveSplatParam = yes
```

Comment:
This is also a pretty big duplicated section.

Reply:
Ok for this one there wouldn't be a good way to fully DRY it up (the existing code still needs to track whether it's seen a splat param yet for purposes of compiling the params), but I trimmed down the comments to be more accurate to that particular context and adjusted the naming here to be more specific to this use case (`haveSplatParam` -> `seenSplatParam`). So made it less of a literal "copy-paste"